### DISCUSSION TITLE
Action to add a commit to a PR

### DISCUSSION THREAD

**Discussion by SoftCreatR** [community/community#24945] [Actions]
<p>Hi,</p>
<p>I’m currently trying to achieve the following:</p>
<p>A user creates a PR, that triggers my action. Within this action, changes are made using a custom script. So far so good, since it works as expected until this point.</p>
<p>Now, I want to create a commit and push it into this PR.</p>
<p>What I’ve found is this action: <a href="https://github.com/marketplace/actions/create-or-update-pull-request" class="inline-onebox" rel="noopener nofollow ugc">Create or Update Pull Request · Actions · GitHub Marketplace · GitHub</a></p>
<p>However, I’m either too stupid to use it right, or it doesn’t work as I’ve expected, because it creates a new PR instead of updating the original one.</p>
<p>Are there alternatives to this action, or is there maybe a way to achieve this without an external action?</p>

**Comment by SoftCreatR** [comment]
<p>Just tried to manually commit &amp; push, but that fails:</p>
<pre><code class="lang-auto">fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:&lt;name-of-remote-branch&gt;
</code></pre>

**Comment by SoftCreatR** [comment, marked as answer]
<p>I was able to make it work, using <code>gh</code>:</p>
<pre><code class="lang-yaml">      - name: Checkout PR
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: gh pr checkout ${{ github.event.pull_request.number }}

      # other stuff

      - name: Push changes
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: git push
</code></pre>
<p>Maybe this helps someone.</p>

**Reply by kimad-cincip** [reply]
This unblocked us, thank you so much!

**Reply by egor-ryashin** [reply]
Day saver

**Reply by heyajulia** [reply]
You're a legend. Thank you!

**Reply by Rkejji** [reply]
Thanks ! btw if someone encounters this error: "failed to run git: fatal: not a git repository (or any of the parent directories): .git" it means that the command "gh checkout pr" was run in a non git repository(doesn't have the .git file). 
To resolve it I added this action in the beginning: actions/checkout@v2.

**Reply by Rkejji** [reply]
Also don't forget to "git add and git commit" in your script before pushing ^^'

**Reply by boclar** [reply]
Thanks! @SoftCreatR

**Comment by sigmaSd** [comment]
It didn't work for me , here is the code I used, it fails to push to the pr with no permission error

https://github.com/sigmaSd/Stimulator/blob/ff3f823227f428497451a8a92956d209536b8b49/.github/workflows/compileTranslations.yml

**Reply by sellout** [reply]
One common reason for this is that if the PR branch is on a fork, rather than in the same repo, `on.pull_request` won’t give you access to secrets (because then a malicious user could open a PR to exfiltrate the secrets).

To get access to the secrets when a PR is on a fork, you need to use `on.pull_request_target`, but the tradeoff is that that then uses the version of the workflow that’s on the branch the PR will be merged _to_, and there are some hurdles to jump through to get the checkout to be the right code for the PR, etc.
